scripts/globals/job_utils/thief.lua

```diff
@@ -0,0 +1,486 @@
+-----------------------------------
+-- Thief Job Utilities
+-----------------------------------
+require('scripts/globals/items')
+require("scripts/globals/msg")
+require("scripts/globals/quests")
+require("scripts/globals/settings")
+require("scripts/globals/status")
+require("scripts/globals/utils")
+-----------------------------------
+xi = xi or {}
+xi.job_utils = xi.job_utils or {}
+xi.job_utils.thief = xi.job_utils.thief or {}
+
+-----------------------------------
+-- Variable Definitions
+-----------------------------------
+
+local despoilDebuffs =
+{
+    xi.effect.EVASION_DOWN,
+    xi.effect.DEFENSE_DOWN,
+    xi.effect.ACCURACY_DOWN,
+    xi.effect.ATTACK_DOWN,
+    xi.effect.MAGIC_ATK_DOWN,
+    xi.effect.MAGIC_DEF_DOWN,
+    xi.effect.SLOW
+}
+
+local stealableSPEffects =
+{
+    xi.effect.MIGHTY_STRIKES,   xi.effect.HUNDRED_FISTS, xi.effect.MANAFONT,     xi.effect.CHAINSPELL,
+    xi.effect.PERFECT_DODGE,    xi.effect.INVINCIBLE,    xi.effect.BLOOD_WEAPON, xi.effect.SOUL_VOICE,
+    xi.effect.MEIKYO_SHISUI,    xi.effect.AZURE_LORE,    xi.effect.TRANCE,       xi.effect.BOLSTER,
+    xi.effect.ELEMENTAL_SFORZO
+}
+
+-- Quadavs Group ID's that the THF AF1 item can be stolen from:
+-- Bronze Quadav groupid = 7949
+-- Garnet Quadav groupid = 7960
+-- Silver Quadav groupid = 7977
+-- Zircon Quadav groupid = 7985
+local validThfQuestMobs =
+{
+    17379367, 17379368, 17379459, 17379470, 17379477, 17379489, 17379493, 17379495, 17379501, 17379505, 17379509,
```

Comment:
This is going to need some love, and a new method in general.  Wall of magic numbers is a show-stopper in my book, even if that's how we used to do things.

Reply:
All of these mobs have the Quadav Stew (item ID 4569) in their drop lists as a stealable item. It looks like the function checks if the mob is one of those particular ID's before it forces a guaranteed steal. My thought is if the `validThfQuestMobs` function checks the drop ID on the mob, and if it matches one of the four drop ID's for the valid mobs it returns true.

Reply:
Perhaps a binding that checks if a certain item is in the droplist for steal?  Seems a bit niche for that, but I think working backwards might be a bit cleaner. 

If <I potentially have stew> (Stew!)

Reply:
`local validThfQuestDropPools =
{
    362, 936, 2254, 2903
}`

and

`local function checkThfAfQuest(player, target)
    if player:getQuestStatus(xi.quest.log_id.WINDURST, xi.quest.id.windurst.THE_TENSHODO_SHOWDOWN) == QUEST_ACCEPTED then
        for i = 1, #validThfQuestDropPools do
            if target:getObjType() == xi.objType.MOB and target:getDropID() == validThfQuestDropPools[i] then
                return true
            end
        end

        return false
    end
end`

Adjusting the functions to the above reduces the number of ID's that have to be tracked. I found only the Beadeaux Bronze, Garnet, Silver, and Zirvon Quadav had those drop lists. I haven't been able to find any core functions that check a mob's drop list for what items are contained in the list.

Reply:
Pulling the elder into this one @TeoTwawki .  Seeing as pools are internal, this seems much more reasonable, but would like his feedback.

Reply:
Oh, wait. `getStealItem` might work, hang on.

Reply:
`getStealItem` does work but the subject Quadav can also have beastcoins stolen from them. Using `getStealItem` randomly pulls from the mob's stealable item pool on their drop list.

Reply:
or, here me out here, we make an onSteal for these the same way we have onCraiticalHit and onmagicHit, and stick that in these mobs scripts.

there is actually a lot I'd like to do on steal attempt / on landing anyway

Reply:
I'm all ears and will happily give it a shot!

Reply:
check `uint16 OnMobWeaponSkillPrepare(CBattleEntity* PMob, CBattleEntity* PTarget)` in luautils.cpp for a decent template to add an "onSteal" event. It looks close enough to what you want.